docs/recipes/WritingTests.md

```diff
@@ -311,46 +303,41 @@ can be tested like:
 
 ```js
 import React from 'react'
-import Enzyme, { shallow } from 'enzyme'
-import Adapter from 'enzyme-adapter-react-16'
+import { render, fireEvent, getByTestId } from '@testing-library/react'
 import Header from '../../components/Header'
 
-Enzyme.configure({ adapter: new Adapter() })
-
-function setup() {
-  const props = {
-    addTodo: jest.fn()
-  }
-
-  const enzymeWrapper = shallow(<Header {...props} />)
-
-  return {
-    props,
-    enzymeWrapper
-  }
-}
-
 describe('components', () => {
   describe('Header', () => {
     it('should render self and subcomponents', () => {
-      const { enzymeWrapper } = setup()
+      const mockAddTodo = jest.fn()
+      const { container, getByText, getByPlaceholder } = render(
+        <Header addTodo={mockAddTodo} />
+      )
+      expect(getByTestId(container, 'header')).toBeInTheDocument()
+      expect(getByText('todos')).toBeInTheDocument()
+      expect(getByPlaceholder('What needs to be done?')).toBeInTheDocument()
```

Comment:
Here's how I'd write this:

```javascript
// the screen variable comes from the React Testing Library import: https://testing-library.com/docs/dom-testing-library/api-queries#screen
render(<Header addTodo={mockAddTodo} />)
expect(screen.getByText(/todos/i)).toBeInTheDocument()
expect(getByPlaceholder(/what needs to be done/i)).toBeInTheDocument()
```

I'd suggest the same kind of changes to the other tests (note the absense of the test ID query, those aren't recommended: https://testing-library.com/docs/guide-which-query)

But actually I don't see much value in this test by itself. These assertions are covered by the other tests... More on this: https://kentcdodds.com/blog/write-fewer-longer-tests

Reply:
I sure agree with you that this test has no meaning, It was just a refactor to the test that was already there so I didn't feel comfortable deleting it :)